pkg/hintrunner/core/hint.go

```diff
@@ -140,6 +140,46 @@ func (hint *TestLessThanOrEqual) Execute(vm *VM.VirtualMachine, _ *hinter.HintRu
 	return nil
 }
 
+type TestLessThanOrEqualAddress struct {
+	dst hinter.CellRefer
+	lhs hinter.ResOperander
+	rhs hinter.ResOperander
+}
+
+func (hint *TestLessThanOrEqualAddress) String() string {
+	return "TestLessThanOrEqualAddress"
+}
+
+func (hint *TestLessThanOrEqualAddress) Execute(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+	lhsPtr, err := hinter.ResolveAsAddress(vm, hint.lhs)
+	if err != nil {
+		return fmt.Errorf("resolve lhs pointer: %w", err)
+	}
+
+	rhsPtr, err := hinter.ResolveAsAddress(vm, hint.rhs)
+	if err != nil {
+		return fmt.Errorf("resolve rhs pointer: %w", err)
+	}
+
+	resFelt := f.Element{}
+	if lhsPtr.SegmentIndex < rhsPtr.SegmentIndex || (lhsPtr.SegmentIndex == rhsPtr.SegmentIndex && lhsPtr.Offset <= rhsPtr.Offset) {
+		resFelt.SetOne()
+	}
```

Comment:
I was thinking it might be a good idea to add a Compare functionality to the MemoryAddress type. Then here you would only have to ask for the result of the comparison, which is more readable.

Reply:
That's a nice suggestion. Made the changes.